### ISSUE TITLE
Getting Error "7/No Match"

### ISSUE THREAD

**Issue by coder4affine** [wenkesj/react-native-voice#159]
After few seconds of start recording getting "7/No Match" Error. I have also tried with sample code and I am getting same error.

My React Native setup is as below

react": "^16.6.3",
"react-native": "^0.57.5",
"react-native-voice": "^0.2.6",

Attached Log details as well as snapshot.
![screenshot 2019-02-22 at 1 25 40 pm](https://user-images.githubusercontent.com/9801499/53227991-df937f00-36a5-11e9-8543-0a764502855a.png)
![screenshot_20190222-132222_get engaged](https://user-images.githubusercontent.com/9801499/53228423-2e8de400-36a7-11e9-980a-b4fece586224.jpg)

**Comment by shivam4ukhandelwal** [comment]
Hey i am also having the same error

**Comment by tunm1228** [comment]
i am also having the same error

**Comment by Goye** [comment]
To solve that problem you need to pass the following config to the .start method:

``` Voice.start('es_US', {
  "RECOGNIZER_ENGINE": "GOOGLE",
   "EXTRA_PARTIAL_RESULTS": true
})``` 

I think the RECOGNIZER_ENGINE is not necessary haha

**Comment by anhnd11** [comment]
same issue in Android devices

**Comment by mrengineer20** [comment]
I am also having the same error!

**Comment by AlkanV** [comment]
i am also getting same error. iOS works seamlessly perfect but on android it throws such error. i put only console.log for every event so that you can see the logs more clearly. 


```
[Tue Mar 02 2021 14:24:26.978]  LOG      services:  ["com.google.android.googlequicksearchbox"]
[Tue Mar 02 2021 14:24:28.464]  LOG      onSpeechStart:  {"error": false}
[Tue Mar 02 2021 14:24:28.578]  LOG      onSpeechVolumeChanged:  {"value": -2}
[Tue Mar 02 2021 14:24:28.817]  LOG      onSpeechVolumeChanged:  {"value": -0.440000057220459}
[Tue Mar 02 2021 14:24:28.849]  LOG      onSpeechVolumeChanged:  {"value": -2}
[Tue Mar 02 2021 14:24:29.360]  LOG      onSpeechVolumeChanged:  {"value": 1.119999885559082}
[Tue Mar 02 2021 14:24:29.710]  LOG      onSpeechVolumeChanged:  {"value": 0.039999961853027344}
[Tue Mar 02 2021 14:24:29.800]  LOG      onSpeechVolumeChanged:  {"value": 0.7599999904632568}
[Tue Mar 02 2021 14:24:29.104]  LOG      onSpeechVolumeChanged:  {"value": -0.440000057220459}
[Tue Mar 02 2021 14:24:29.113]  LOG      onSpeechVolumeChanged:  {"value": 0.8799998760223389}
[Tue Mar 02 2021 14:24:29.210]  LOG      onSpeechVolumeChanged:  {"value": 0.16000008583068848}
[Tue Mar 02 2021 14:24:29.212]  LOG      onSpeechVolumeChanged:  {"value": 1}
[Tue Mar 02 2021 14:24:29.212]  LOG      onSpeechVolumeChanged:  {"value": 10}
[Tue Mar 02 2021 14:24:29.469]  LOG      onSpeechVolumeChanged:  {"value": -2}
[Tue Mar 02 2021 14:24:33.729]  LOG      onSpeechError:  {"error": {"message": "7/No match"}}
```

and this is my start function with options: 

```
        Voice.start('en-US', {
          RECOGNIZER_ENGINE: 'services',
          EXTRA_PARTIAL_RESULTS: true,
        })
```

any help will be appreciated, thank you

**Comment by TruongLCC** [comment]
yep, I get same err too. It is OK in fist time, and return an err in next time. I am using their example code.

**Comment by Nikita279** [comment]
I'm facing the same issue. Did anyone find a solution to this?

**Comment by JJ810** [comment]
Hey guys! I am also having this issue! Does anyone have any solution for this?

**Comment by learncodingforweb** [comment]
Getting same issue on android

**Comment by lefnire** [comment]
Same issue. Either `5/Client side error` or `7/No match`, very frequent and I can't spot a pattern. Sometimes it will try to listen for a bit, other times it will be immediate on `Voice.start()`

**Comment by khaihoan2306** [comment]
try running on API level 30, I cant run it on 31 but working on API 30

**Comment by lefnire** [comment]
@khaihoan2306 I'm on 30. Thanks for the possibility though, was worth taking a look (I forgot which I was on). It's not a certain thing; it's like 50/50 (successful recognition vs 5, 7, or 8). Where I left off last year, my hunch was that there's some sort of race condition on the assigned functions. Eg, recognize/start/stop competing; possibly if it picks up background noise against the speaker.

**Comment by DasSumana558025** [comment]
I am facing the same issue , Does anyone find any solution

**Comment by Vishal-Isharani** [comment]
I'm also facing the same issue.

**Comment by Sanath91009** [comment]
Any updates on this issue ? @imyagnesh

**Comment by Jfmetcalf5** [comment]
Any updates?  I just came across this issue as well

**Comment by w3ever** [comment]
Any updates?

**Comment by hgamage** [comment]
im also facing this issue

**Comment by amalraj7980** [comment]
i also facing the same issue

**Comment by lucadev22** [comment]
be sure to have checked the "Virtual microphone uses hosta audio input" option, in Microphone options from the android emulator

**Comment by WillGITCode** [comment]
> be sure to have checked the "Virtual microphone uses hosta audio input" option, in the Microphone options from the android emulator
@lucadev22 That was it, thanks!

**Comment by branchdev98** [comment]
Any update? I have same error!

**Comment by peterhck** [comment]
Has no one been able to solve this issue?

**Comment by inders-agsft** [comment]
If you are using android emulator, make sure to tick  Microphone => "Virtual microphone uses host audio input" which you can get by clicking "..." in Running Devices.  If must work fine on real device.

I was using emulator and for me checking "Virtual microphone uses host audio input" solved it. Although a lot of delay in detecting voice but was good enough to confirm that it works :-)

**Comment by aniketjha9179** [comment]
> If you are using android emulator, make sure to tick Microphone => "Virtual microphone uses host audio input" which you can get by clicking "..." in Running Devices. If must work fine on real device.
> 
> I was using emulator and for me checking "Virtual microphone uses host audio input" solved it. Although a lot of delay in detecting voice but was good enough to confirm that it works :-)

bro you really solved my issue I just turned my microphone from emulator and all is good now . I was getting this error since yesterday : LOG  speech error:  {"error": {"message": "7/No match"}}
now all is good let's meet somewhere else on some different issue 🤝🧡

**Comment by yahiaslim12** [comment]
has no one able to solve this error ! ;(

**Comment by Muhammad-Nafees** [comment]
@lucadev22 do you know how to solve this problem in Android Real Device not emulator?

**Comment by abhay-keyvalue** [comment]
Any solution for this issue?? 
"Virtual microphone uses hosta audio input" option ticked but not fix not worked for me

**Comment by m2001said** [comment]
![image](https://github.com/user-attachments/assets/a40f2cbd-bcbb-48ee-9bcf-d5210d08415d) 

this is the solution for me and it worked well

**Comment by jernland0279** [comment]
> recognize

Thanks a lot, it works!

**Comment by jernland0279** [comment]
You're right. It works for me, thanks.

**Comment by hassnain0** [comment]
Thanks, that worked for me

**Comment by mkamals1989** [comment]
Yes, the issue still happens in Android real device. I get both the error

**Comment by lefnire** [comment]
Y'all this is 7 yrs old, it ain't gonna get fixed. Go native if you need this

**Comment by safaiyeh** [comment]
This project has been deprecated and archived. Please migrate to [`expo-speech-recognition`](https://github.com/jamsch/expo-speech-recognition). Thank you for your contributions! ❤️
